packages/ui/src/components/notifications/NotificationCenter.test.tsx

```diff
@@ -0,0 +1,169 @@
+        const { queryByText } = renderComponent();
+
+        waitFor(() => {
+            emitEvent('test-1');
+            emitEvent('test-2');
+            emitEvent('test-3');
+            emitEvent('test-4');
+            emitEvent('test-5');
+            emitEvent('test-6');
+        });
+
+        expect(queryByText('test-1')).not.toBeInTheDocument();
+        expect(queryByText('test-2')).toBeInTheDocument();
+        expect(queryByText('test-3')).toBeInTheDocument();
+        expect(queryByText('test-4')).toBeInTheDocument();
+        expect(queryByText('test-5')).toBeInTheDocument();
+        expect(queryByText('test-6')).toBeInTheDocument();
+    });
+
+    it('should hide notifications after the hideAfter prop', () => {
+        const { queryByText } = renderComponent({ hideAfter: 1000 });
+
+        emitEvent('test-1');
+
+        expect(queryByText('test-1')).toBeInTheDocument();
+
+        act(() => {
+            vi.advanceTimersByTime(1000);
+        });
+
+        waitFor(() => {
+            expect(queryByText('test-1')).not.toBeInTheDocument();
+        });
+    });
+
+    it('should hide notifications after the cacheTimeout prop', () => {
+        const { queryByText, getByTestId } = renderComponent({ cacheTimeout: 1000 });
+
+        emitEvent('test-1');
+        emitEvent('test-2');
+        emitEvent('test-3');
+
+        expect(queryByText('test-1')).toBeInTheDocument();
+        expect(queryByText('test-2')).toBeInTheDocument();
+        expect(queryByText('test-3')).toBeInTheDocument();
+
+        act(() => {
+            vi.advanceTimersByTime(1000);
+        });
+
+        waitFor(() => {
+            expect(getByTestId('notification-center')).toBeEmptyDOMElement();
+        });
+    });
+
+    it("should close the notification when it's onClose prop is called", () => {
+        const { queryByText, getByText } = renderComponent();
+
+        emitEvent('test-1');
+
+        expect(queryByText('test-1')).toBeInTheDocument();
+
+        act(() => {
+            fireEvent.click(getByText('test-1'));
+        });
+
+        waitFor(() => {
+            expect(queryByText('test-1')).not.toBeInTheDocument();
+        });
+    });
+});
+
+describe('notify', () => {
+    it('should dispatch a CustomEvent', () => {
+        const eventListener = vi.fn();
+        document.addEventListener(NOTIFICATION_EVENT, eventListener);
+
+        notify({ type: 'info', message: 'test' });
+
+        expect(eventListener).toHaveBeenCalled();
+    });
+
+    it('should dispatch a CustomEvent with a random UUID', () => {
+        const eventListener = vi.fn();
+        document.addEventListener(NOTIFICATION_EVENT, eventListener);
+        const mockLocalRandomUUID = vi.fn(() => 'test-uuid');
+        global.crypto = {
+            ...global.crypto,
+            randomUUID: mockLocalRandomUUID as () => `${string}-${string}-${string}-${string}-${string}`,
```

Comment:
we can create the type ```type UUID = `${string}-${string}-${string}-${string}-${string}`;``` and have the mock function return something like `test-test-test-test-test`. In this way we won't have to cast `test-uuid` to something that it's not